src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1396,6 +1397,14 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
                             ADALError.DEVELOPER_ACTIVITY_IS_NOT_RESOLVED));
                 }
             } else {
+                Map<String, String> eventProperties = new InstrumentationPropertiesBuilder(request, result)
+                        .add(InstrumentationIDs.IS_BROKER_APP, Boolean.TRUE.toString())
+                        .add(InstrumentationIDs.ERROR_CLASS,
+                                result == null ? InstrumentationIDs.AUTH_RESULT_EMPTY : InstrumentationIDs.NO_ACCESS_TOKEN_IN_RESPONSE)
+                        .build();
+                ClientAnalytics.logEvent(
+                        InstrumentationIDs.AUTH_TOKEN_NOT_RETURNED,
```

Comment:
Irina that would be hard to use - let's use event name "TokenRequest" and two properties "Result":"Failed", "Reason":"AUTH_TOKEN_NOT_RETURNED". 

Overall do not create - event type for each condition. Instead design event types around scenario.